cpp/dspace/doi.cc

```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+#include <regex>
+#include <set>
+#include <unordered_map>
+#include <cstdlib>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "Downloader.h"
+#include "EmailSender.h"
+#include "HttpHeader.h"
+#include "JSON.h"
+#include "StlHelpers.h"
+#include "StringUtil.h"
+#include "VuFind.h"
+#include "util.h"
+
+using namespace std;
+
+const std::string DSPACE_SERVER_URL("https://publikationen.uni-tuebingen.de/rest/items/");
+const std::string DOI_KEY("dc.relation.uri");
+const std::string PUBLIKATION_TITLE_KEY("dc.title");
+unsigned int INSTANCE_COUNT(15000);
+const std::vector<std::string> NOTIFICATION_EMAIL_ADRESSES = { "andrii.lysohor@uni-tuebingen.de" };
+
+void SendNotificationsForDOI(const std::string email_subject, const std::string email_message) {
+    if (EmailSender::SimplerSendEmail("no-reply@ub.uni-tuebingen.de", NOTIFICATION_EMAIL_ADRESSES, email_subject, email_message,
+                                      EmailSender::MEDIUM)
+        > 299)
+        LOG_ERROR("Failed to send the DOI notification email!");
+}
+
+void UpdateItem(DbConnection * const db_writer, const std::string publication_id) {
+    db_writer->queryOrDie("UPDATE tuefind_publications SET doi_notification = NOW() WHERE id= " + publication_id + "");
+}
+
+std::string ParseDataFromXML(std::shared_ptr<const JSON::ArrayNode> top_node_array, const std::string fild) {
```

Comment:
`fild` seems to be a typo, do you mean `field`?
and it should be a reference => `const std::string &field`